src/ast/passes/codegen_llvm.cpp

```diff
@@ -3212,19 +3212,19 @@ Value *CodegenLLVM::createLogicalAnd(Binop &binop)
   Value *lhs;
   auto scoped_del_left = accept(binop.left);
   lhs = expr_;
-  b_.CreateCondBr(
-      b_.CreateICmpNE(lhs, b_.GetIntSameSize(0, lhs), "lhs_true_cond"),
-      lhs_true_block,
-      false_block);
+  Value *lhs_zero_value = Constant::getNullValue(expr_->getType());
```

Comment:
nit: IMO reads a bit more clear to use `lhs`. 

same for below occurrences 
```suggestion
  Value *lhs_zero_value = Constant::getNullValue(lhs->getType());
```

Reply:
Done.